programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -37,17 +35,18 @@ pub struct DepositV3<'info> {
 
     #[account(
         seeds = [b"route", input_token.as_ref(), state.key().as_ref(), destination_chain_id.to_le_bytes().as_ref()],
-        bump
+        bump,
+        constraint = route.enabled @ CommonError::DisabledRoute
```

Comment:
this was explicit check in EVM implementation. why move it here in constraints?

Reply:
I moved it to match how we apply similar constraints on `fill.rs` https://github.com/across-protocol/contracts/blob/5c1dc6ed4ecb0bf87430730b91e91a26f11efad8/programs/svm-spoke/src/instructions/fill.rs#L27

Reply:
For fills EVM implementation has `unpausedFills` modifier, so we moved it to constraints so that implementation logic is as close as possible in SVM

Reply:
ye, I recall this. that still feels funny to me though. Would we not rather be consistent between rust implementation? the EVM logic feels inconsistent between the modifier use vs inline checks anyways.

Reply:
I'm happy to revert it though and safe it for later or keep it like this